handout/modules/lazy-loading-module.md

```diff
@@ -0,0 +1,152 @@
+      <a routerLink="lazy">Lazy</a>
+    </nav>
+    <router-outlet></router-outlet>
+  `
+})
+export class AppComponent {}
+```
+
+Our navigation system has only two paths: `eager` and `lazy`. To know what those paths are loading when clicking on them we need to take a look at the `routing` object that we passed to the root module.
+
+_app/app.routing.ts_
+
+```js
+import { ModuleWithProviders } from '@angular/core';
+import { Routes, RouterModule } from '@angular/router';
+
+import { EagerComponent } from './eager.component';
+
+const routes: Routes = [
+  { path: '', redirectTo: 'eager', pathMatch: 'full' },
+  { path: 'eager', component: EagerComponent },
+  { path: 'lazy', loadChildren: 'lazy/lazy.module#LazyModule' }
+];
+
+export const routing: ModuleWithProviders = RouterModule.forRoot(routes);
+```
+
+Here we can see that the default path in our application is called `eager` which will load `EagerComponent`.
+
+_app/eager.component.ts_
+
+```js
+import { Component } from '@angular/core';
+
+@Component({
+  template: '<p>Eager Component</p>'
+})
+export class EagerComponent {}
+```
+
+But more importantly, we can see that whenever we try to go to the path `lazy`, we are going to lazy load a module called conveniently `LazyModule`. Look closely at the definition of that route:
+
+```js
+{ path: 'lazy', loadChildren: 'lazy/lazy.module#LazyModule' }
+```
+
+There's a few important things to notice here:
+
+1. We are using the property `loadChildren` instead of `component`.
+2. We are passing a string instead of a symbol to avoid loading the module eagerly.
+3. We are defining not only the path to the module but the name of the class as well.
+
+There's nothing special about `LazyModule` other than it has its own `routing` and a component called `LazyComponent`.
+
+_app/lazy/lazy.module.ts_
+
+```js
+import { NgModule } from '@angular/core';
+
+import { LazyComponent }   from './lazy.component';
+import { routing } from './lazy.routing';
+
+@NgModule({
+  imports: [routing],
+  declarations: [LazyComponent]
+})
+export class LazyModule {}
+```
+
+> If we define the class `LazyModule` as the `default` export of the file, we don't need to define the class name in the `loadChildren` property as shown above.
+
+The `routing` object is very simple and only defines the default component to load when navigating to the `lazy` path. 
+
+_app/lazy/lazy.routing.ts_
+
+```js
+import { ModuleWithProviders } from '@angular/core';
+import { Routes, RouterModule } from '@angular/router';
+
+import { LazyComponent } from './lazy.component';
+
+const routes: Routes = [
+  { path: '', component: LazyComponent }
+];
+
+export const routing: ModuleWithProviders = RouterModule.forChild(routes);
+```
+
+Notice that we are using the method call `forChild` instead of `forRoot` to create the routing object. We should always do that when creating a routing object for a feature module, no matter if the module is supposed to be eagerly or lazy loaded.
```

Comment:
- `we are using` to `we use` (style guide active voice)
- `no matter if the module is supposed to be eagerly or lazy loaded` to `no matter if the module is supposed to be loaded eagerly or lazily`